### ISSUE TITLE
TileMap Physics layer won't disable when process set to disable while every other physics object will

### ISSUE THREAD

**Issue by EviTRea** [godotengine/godot#86527]
### Tested versions

v4.2.1.stable.flathub [b09f793f5]

### System information

Godot v4.2.1.stable (b09f793f5) - Freedesktop SDK 23.08 (Flatpak runtime) - X11 - GLES3 (Compatibility) - NVIDIA GeForce RTX 3070 Ti (nvidia) - AMD Ryzen 5 5600X 6-Core Processor (12 Threads)

### Issue description

If we put a RayCast2D and make it intersect with any disabled CollisionObject2D, it won't detect it;
However, a disabled TileMap WILL be detected by Area2d and RayCast2d.

(The rest is just rant, I know it's not appropriate but I'm so upset)
I've been trying to work around this for a month now, I've wasted weeks on converting TileMap data into arrays and painting the map on codes with more failure.
It used to work with simple hiding and showing, and apparently it was a bug, after an update now invisible object can also block your way!! Horray!!
But now I've notice there's inconsistency between TileMap and other physics object, I hereby claim my code has no issue, I'll put this part of the project on halt until it's fixed.

### Steps to reproduce

1.Set up any CollisionObject2D inherited class except PhysicalBone2D, give them CollisionShape2D to make them functional
2.Also create a TileMap with Physics layer property
3.Create RayCast2D or Area2D, print raycast.get_collider() or area.get_overlapping_bodies()/get_overlapping_areas() to check if all of the above are detected
4.Set everything except detector's process mode to disable
5.Print again, you should see TileMap is still there while every other stuffs are undetectable 

### Minimal reproduction project (MRP)

[ScriptTest.zip](https://github.com/godotengine/godot/files/13772349/ScriptTest.zip)
1.Press "Detect" and see how much it detect
2.Press "Set Disable" to disable everything aside detector
3.Press "Detect" again to see how the TileMap stands his ground to block my project progression
(Note: I originally had another "Set Enable" button, but somehow only CharacterBody agrees to turn back on, might be another issue but I'm not bothering)

**Comment by AThousandShips** [comment]
> (The rest is just rant, I know it's not appropriate but I'm so upset)

Then please don't rant, see the [code of conduct](https://godotengine.org/code-of-conduct/), if you know it's not appropriate then please be professional and act accordingly, knowingly breaking a rule isn't very professional or acceptable...

**Comment by EviTRea** [comment]
> > (The rest is just rant, I know it's not appropriate but I'm so upset)
> 
> Then please don't rant, see the [code of conduct](https://godotengine.org/code-of-conduct/), if you know it's not appropriate then please be professional and act accordingly, knowingly breaking a rule isn't very professional or acceptable...
> 
Funnily enough, 10 mins after posting this, a perfect solution stroked me...
Just move everything not in need out of the frame and move them back when needed, oldest trick in the gamedev, should've been first thing to try in hindsight

Yeah, I'm calm now, I've edited the post, sorry about it and I'll find a better way to handle frustration next time

**Comment by kaluluosi** [comment]
It seems that TileMapLayer inherits RefCounted not Node, so it dont have process mode, it runs independently out of scenetree.
Then Tilemap doesn't control TileMapLayer  according to it's Process Mode.
The only way to disable it is this.
![image](https://github.com/godotengine/godot/assets/1620585/0d05b65c-7d90-4f9e-8efe-c5bef536522c)

Disable the layer not tilemap.

I believe that the process mode behavior of the TileMapLayer should be consistent with its parent TileMap node, which is reasonable.

**Comment by AThousandShips** [comment]
@kaluluosi Please do not link to issues in commits, it creates a lot of noise that makes it hard to follow the conversation

**Comment by kaluluosi** [comment]
> @kaluluosi Please do not link to issues in commits, it creates a lot of noise that makes it hard to follow the conversation

My apologize.
